spec/services/creators/user_creator_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe UserCreator do
+  describe '.create!' do
+    it 'calls :create! on an instance of UserCreator' do
+      params = instance_spy('Hash')
+      user_creator = mock_user_creator(params)
+      described_class.create!(params)
+      expect(user_creator).to have_received(:create!)
+    end
+  end
+
+  describe '#create!' do
+    context 'success' do
+      it 'returns a valid user' do
+        params = instance_spy('ActionController::Parameters',
+                              to_h: valid_params)
+        result = described_class.create!(params)
+        expect(result[:object]).to be_an_instance_of(User)
+      end
+      it 'returns a success message' do
+        params = instance_spy('ActionController::Parameters',
+                              to_h: valid_params)
+        result = described_class.create!(params)
+        expect(result[:msg]).to have_key(:success)
+      end
+      it 'sends a confirmation e-mail' do
+        params = instance_spy('ActionController::Parameters',
+                              to_h: valid_params)
+        mailer = instance_spy('user_mailer')
+        described_class.new(params, mailer).create!
+        expect(mailer).to have_received(:new_user_confirmation)
+      end
+
+      context 'without CAS' do # rubocop:disable RSpec/NestedGroups
+        before { allow(User).to receive(:cas_auth?).and_return(false) }
+        it 'assigns a random password to the user' do
+          params = instance_spy('ActionController::Parameters',
+                                to_h: valid_params)
+          result = described_class.create!(params)
+          expect(result[:object].password).not_to be_empty
+        end
+      end
+      # TODO: this test currently fails because stubbing out User.cas_auth?
+      # doesn't change how the User class was loaded (without CAS), and so it's
+      # still running password validations. Even switching to any_instance_of
+      # doesn't help because it has to do with when Rails loads the User
+      # class... not sure what to do.
+      context 'with CAS' do # rubocop:disable RSpec/NestedGroups
```

Comment:
I'm not sure how to make this test pass, it may not be possible since we'd have to reload the User class. Thoughts?